common/adder/adder.xml

```diff
@@ -0,0 +1,103 @@
+<?xml version="1.0" encoding="utf-8"?>
+<component version="1.4" xmlns="http://schemas.circuit-diagram.org/circuitDiagramDocument/2012/component/xml">
+  <declaration>
+    <meta name="name" value="Adder" />
+    <meta name="minsize" value="50" />
+    <meta name="author" value="Corsaka" />
+    <meta name="guid" value="8f764035-18c6-4421-ac7f-5cbd327d4e78" />
+
+    <property name="Type" type="enum" default="Full" serialize="t" display="Type">
+      <option>Full</option>
+      <option>Half</option>
+    </property>
+
+    <configurations>
+      <configuration name="Full Adder" value="Type:Full" implements="full" />
+      <configuration name="Half Adder" value="Type:Half" implements="half" />
+    </configurations>
+  </declaration>
+  <connections>
+    <group conditions="horizontal">
+      <connection name="A" start="_Start-7x-30y" end="_Start-7x-60y" edge="End" />
+      <connection name="B" start="_End+7x-30y" end="_End+7x-60y" edge="End" />
```

Comment:
The `A` and `B` connections aren't appearing because they need to be aligned to the 10x10 grid (they appear as crosses when _debug layout_ is enabled).

![image](https://user-images.githubusercontent.com/5303088/75492234-44677080-59af-11ea-85bf-77490330b35a.png)

Reply:
Which appear as crosses; Good connections or bad connections? It doesn't look like A or B are touched there.

Reply:
Each cross represents a good connection. The "bad connections" are not shown at all. So each cross represents a point that a wire/another component can be connected to (i.e. would have a circle connection point drawn there if you did so).

Each cross also has a line next to it to represent how you could connect a wire to that point. So for the small 1 grid square lines, those are "non-edge" connections that you could connect the end of a wire to. The long 4 grid square lines are "edge" connections that you could connect the middle of a wire to.